src/tls/main.c

```diff
@@ -0,0 +1,146 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2019 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <argp.h>
+#include <err.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <signal.h>
+
+#include <common/cockpitwebcertificate.h>
+#include "utils.h"
+#include "server.h"
+
+#define COCKPIT_WS PACKAGE_LIBEXEC_DIR "/cockpit-ws"
+
+/* CLI arguments */
+struct arguments {
+  uint16_t port;
+  int no_tls;
+  unsigned idle_timeout;
+};
+
+#define OPT_NO_TLS 1000
+#define OPT_IDLE_TIMEOUT 1001
+
+static unsigned
+arg_parse_unsigned (char *arg, struct argp_state *state, unsigned min, unsigned max, const char *error_msg)
+{
+  char *endptr = NULL;
+  long num = strtol (arg, &endptr, 10);
+
+  if (!*arg || *endptr != '\0' || num < min || num > max)
```

Comment:
Comparing a signed long to an unsigned int has different behaviour on 32bit and 64bit.  I think in this particular case, since you use small `max`, you'll be ok, but still *shudder*.

I'd consider changing min and max to be signed.